hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/ozoneimpl/TestOnDemandContainerDataScanner.java

```diff
@@ -277,19 +278,18 @@ public void testUnhealthyContainerNotRescanned() throws Exception {
         .setState(UNHEALTHY);
     // Update the mock to reflect this.
     when(unhealthy.getContainerState()).thenReturn(UNHEALTHY);
-    assertFalse(unhealthy.shouldScanData());
+    assertTrue(unhealthy.shouldScanData());
 
     // Clear metrics to check the next run.
     metrics.resetNumContainersScanned();
     metrics.resetNumUnhealthyContainers();
```

Comment:
Even though this is in the original, I think it would be better to not clear the metrics on each run, and instead check the accumulated values (same for all the tests). This is similar to how it would work when actually running. This also exposes something else we should update:

The numUnhealthyContainers metric should be a count of how many unhealthy containers are on the datanode. Currently an existing unhealthy container will be double counted when the scanner sees it again. We should update the scanner so that if the container's state is already unhealthy, we do not increment this metric.